### PR TITLE
Close the socket connection if we don't read from it

### LINKED ISSUE #640: statObject hangs process until socket times out
If statObject() is called successfully, the client does not seem to clean up its sockets correctly.  This  leads to the entire process hanging for ~15 seconds until the socket times out.

### PR DESCRIPTION
**Pull request by krishnasrinivas**
fixes #640

### CODE REVIEW

**Comment by kannappanr** [comment]
Travis failed

File: src/main/minio.js [hunk 1]
```diff
@@ -324,6 +324,9 @@ export class Client {
     if (!isString(region)) {
       throw new TypeError('region should be of type "string"')
     }
+    if ((returnResponse !== true) && (returnResponse !== false)) {
```
Comment (brendanashworth) [inline src/main/minio.js]: Would `typeof returnResponse != 'boolean'` be better here?

Comment (harshavardhana) [reply src/main/minio.js]: We have a functions isBoolean for this purpose.

Comment (krishnasrinivas) [reply src/main/minio.js]: Ah yes, will use that

**Pull request merged**